agent/base

```diff
@@ -160,6 +160,7 @@ function get_redhat_version() {
 }
 
 function check_install_rpm {
+    [ "$PBENCH_SKIP_CHECK_INSTALL_RPM" == true ] && return
```

Comment:
Hello @webbnh, I'm actually struggling with `check_install_rpm` on many fronts. Previously I was playing with pbench on non-rpm-based systems, but I do understand that is not the primary usecase right now.

I also had to implement way to supply the correct version for fio and uperf on systems that comes with different version out of the box (I'm only comparing the same distros together so I don't really care about the different uperf version and besides I get notified of a different rpm version if that comparison happens). On older versions I'm also compiling fio myself (to get nbd backend), so the version logged by pbench is actually different than the one used via `benchmark_bin` (I'm treating this myself to get the used version instead).

As for linpack, it's even trickier as the hardcoded (not configurable) linpack version is `pbench-linpack-11.1.3.x86_64` and this version is even used for data-file location, while the only available version is an updated `pbench-linpack-11.1.3-1.x86_64`. Anyway the data-file location stayed the same (it's a minor update) so updating the version actually fails (because the data-file location based on rpm version differs) and relying on it also fails as it's not the hardcoded version.

For me I don't really need the rpm check as I just want to the version that is installed or specified via custom path. I'd welcome, though, to log the version based on the benchmark binary (`fio --version`).

Reply:
> I also had to implement way to supply the correct version for fio and uperf on systems that comes with different version out of the box (I'm only comparing the same distros together so I don't really care about the different uperf version and besides I get notified of a different rpm version if that comparison happens). On older versions I'm also compiling fio myself (to get nbd backend), so the version logged by pbench is actually different than the one used via `benchmark_bin` (I'm treating this myself to get the used version instead).

@ldoktor, can you file an issue for this?

We don't have much support for non-RPM based linux distributions (see https://github.com/distributed-system-analysis/pbench/issues/91).  Any help towards that would be welcome.

As part of the above issue you open, we could do work to replace the RPM checks with simple path execution checks which are independent.

> As for linpack, it's even trickier as the hardcoded (not configurable) linpack version is `pbench-linpack-11.1.3.x86_64` and this version is even used for data-file location, while the only available version is an updated `pbench-linpack-11.1.3-1.x86_64`. Anyway the data-file location stayed the same (it's a minor update) so updating the version actually fails (because the data-file location based on rpm version differs) and relying on it also fails as it's not the hardcoded version.

Let's take the step of changing `pbench-linpack` to not hard-code the version and move it to the `pbench-agent-default.cfg` file, with a minimum supported version, like we did for `fio`, `uperf`, etc.

> 
> For me I don't really need the rpm check as I just want to the version that is installed or specified via custom path. I'd welcome, though, to log the version based on the benchmark binary (`fio --version`).

Tracking that request in an issue so that we could schedule the work, or accept a community contribution, would be great.

Reply:
Regardless, I'd like to avoid this kind of change in this PR and focus on the functional enhancements ported over from pbench-run-benchmark.

Reply:
OK, let me re-send this separately. Anyway let me keep it here as well as without it there is no way to check this (the pbench-linpack of the desired version does not exists)